comps/llms/README.md

```diff
@@ -6,111 +6,123 @@ A prerequisite for using this microservice is that users must have a LLM text ge
+export MICROSERVICE_DIR=your_chosen_microservice
+
+pip install -r ${OPEA_GENAICOMPS_ROOT}/comps/llms/text-generation/${MICROSERVICE_DIR}/requirements.txt
 ```
 
-### 1.2 Start LLM Service
+### 1.2 Start LLM Service with Python Script
 
-#### 1.2.1 Start TGI Service
+#### 1.2.1 Start the TGI Service
 
 ```bash
-export HF_TOKEN=${your_hf_api_token}
-export LANGCHAIN_TRACING_V2=true
-export LANGCHAIN_API_KEY=${your_langchain_api_key}
-export LANGCHAIN_PROJECT="opea/gen-ai-comps:llms"
-docker run -p 8008:80 -v ./data:/data --name tgi_service --shm-size 1g ghcr.io/huggingface/text-generation-inference:1.4 --model-id ${your_hf_llm_model}
+export TGI_LLM_ENDPOINT="http://${your_ip}:8008"
+python ${OPEA_GENAICOMPS_ROOT}/comps/llms/text-generation/tgi/llm.py
 ```
 
-#### 1.2.2 Start vLLM Service
+#### 1.2.2 Start the vLLM Service
 
 ```bash
-export HUGGINGFACEHUB_API_TOKEN=${your_hf_api_token}
-docker run -it --name vllm_service -p 8008:80 -e HF_TOKEN=${HUGGINGFACEHUB_API_TOKEN} -v ./data:/data opea/vllm:cpu /bin/bash -c "cd / && export VLLM_CPU_KVCACHE_SPACE=40 && python3 -m vllm.entrypoints.openai.api_server --model ${your_hf_llm_model} --port 80"
+export vLLM_LLM_ENDPOINT="http://${your_ip}:8008"
+python ${OPEA_GENAICOMPS_ROOT}/comps/llms/text-generation/vllm/llm.py
 ```
 
-### 1.2.3 Start Ray Service
+#### 1.2.3 Start the Ray Service
 
 ```bash
-export HUGGINGFACEHUB_API_TOKEN=${your_hf_api_token}
-export TRUST_REMOTE_CODE=True
-docker run -it --runtime=habana --name ray_serve_service -e OMPI_MCA_btl_vader_single_copy_mechanism=none --cap-add=sys_nice --ipc=host -p 8008:80 -e HUGGINGFACEHUB_API_TOKEN=$HUGGINGFACEHUB_API_TOKEN -e TRUST_REMOTE_CODE=$TRUST_REMOTE_CODE opea/llm-ray:latest /bin/bash -c "ray start --head && python api_server_openai.py --port_number 80 --model_id_or_path ${your_hf_llm_model} --chat_processor ${your_hf_chatprocessor}"
+export RAY_Serve_ENDPOINT="http://${your_ip}:8008"
+python ${OPEA_GENAICOMPS_ROOT}/comps/llms/text-generation/ray_serve/llm.py
 ```
 
-### 1.3 Verify the LLM Service
+## 🚀2. Start Microservice with Docker (Option 2)
 
-#### 1.3.1 Verify the TGI Service
+You can use either a published docker image or build your own docker image with the respective microservice Dockerfile of your choice.
 
-```bash
-curl http://${your_ip}:8008/generate \
-  -X POST \
-  -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":17, "do_sample": true}}' \
-  -H 'Content-Type: application/json'
-```
+### 2.1 Start LLM Service with published image
 
-#### 1.3.2 Verify the vLLM Service
+#### 2.1.1 Start TGI Service
 
 ```bash
-curl http://${your_ip}:8008/v1/completions \
-  -H "Content-Type: application/json" \
-  -d '{
-  "model": ${your_hf_llm_model},
-  "prompt": "What is Deep Learning?",
-  "max_tokens": 32,
-  "temperature": 0
-  }'
-```
-
-#### 1.3.3 Verify the Ray Service
+export HF_LLM_MODEL=${your_hf_llm_model}
```

Comment:
Do you need to explain what "your_hf_llm_model" and "your_hf_api_token" are?  Is it obvious that hf refers to HuggingFace?

Reply:
Yes I would say so, because anyone who are familiar with ML frameworks would probably know some acronyms, like TF-Tensorflow, PT-PyTorch and HF-HuggingFace. But as you said, it doesn't harm to mention somewhere too.